### PR TITLE
Switch to reusable code quality action in CI

### PR DESCRIPTION
**Pull request by didrocks**
This is the first phase of more reusable Go and general CI. Tackling Quality/Sanity checks.

UDENG-599

### CODE REVIEW

File: .github/workflows/qa.yaml [hunk 1]
```diff
@@ -150,59 +128,3 @@ jobs:
         if: startsWith(github.ref, 'refs/tags/')
         with:
           files: installer/Output/adwatchd_setup.exe
-
-  generators:
-    name: Generated files are up to date
```
Comment (denisonbarbosa) [inline .github/workflows/qa.yaml:155]: Are we dropping this entirely?

File: .github/workflows/qa.yaml [hunk 2]
```diff
@@ -25,33 +25,11 @@ jobs:
...
-        with:
-          go-version-file: go.mod
-      - name: Get version of golanci-lint to use
-        id: cilint-version-fetch
-        # This handles "require foo version" and "require (\nfoo version\n)"" formats
-        run: echo "version=$(grep golangci-lint tools/go.mod | rev | cut -f1 -d' ' | rev)" >> $GITHUB_OUTPUT
-      - name: Code formatting, vet, static checker Security…
-        uses: golangci/golangci-lint-action@v3
+      - name: Go code sanity check
+        uses: canonical/desktop-engineering/gh-actions/go/code-sanity@main
```
Comment (denisonbarbosa) [inline .github/workflows/qa.yaml:29]: That's quite some nesting...

File: .github/workflows/qa.yaml (hunk 1 above)
Comment (didrocks) [reply .github/workflows/qa.yaml:155]: Look at the reused action. It has the generator :)

Comment (denisonbarbosa) [reply .github/workflows/qa.yaml:155]: Oh, nice!

**Review by denisonbarbosa — APPROVED** [review]
Looks great and clean! Very nice to glob everything in a standardized action. I'm a bit worried about a single job doing everything, but we can see how it goes!

**Comment by didrocks** [comment]
Thanks! The job is gathering the results. I will demonstrate the outcome once everyone is back :)

**Pull request merged**